db/import.go

```diff
@@ -147,7 +147,7 @@ func (db *DatabaseCollectionWithUser) importDoc(ctx context.Context, docid strin
 		existingDoc.Expiry = *expiry
 	}
 
-	docOut, _, err = db.updateAndReturnDoc(ctx, newDoc.ID, true, expiry, mutationOptions, existingDoc, func(doc *Document) (resultDocument *Document, resultAttachmentData AttachmentData, createNewRevIDSkipped bool, updatedExpiry *uint32, resultErr error) {
+	docOut, _, err = db.updateAndReturnDoc(ctx, newDoc.ID, true, expiry, mutationOptions, existingDoc, true, func(doc *Document) (resultDocument *Document, resultAttachmentData AttachmentData, createNewRevIDSkipped bool, updatedExpiry *uint32, resultErr error) {
```

Comment:
I am about to make a change to do something similar but add `metadataOnly` as a return value to the callback function.

This code is wrong here because would mark a document that has inline sync metadata as a metadata only update.

Reply:
I didn't think the callback function was the right scope, because of the requirement for the additional interaction with documentUpdateFunc, and it felt like that obfuscated the decision a bit.  Metadata migration for inline sync data is performed in a separate kv op, so I don't think it's a consideration here.

Reply:
I now understand why this is different from metadata only updates and I'll have to test this when I implement a `_mou` xattr. I didn't understand that `importDoc` will call `migrateMetadata` and it will update the body from what was originally passed in.